core/runtime/src/main/java/io/quarkus/runtime/configuration/ConfigInstantiator.java

```diff
@@ -60,7 +61,16 @@ public static void handleObject(Object o) {
         }
 
         final Class<?> cls = o.getClass();
-        final String name = dashify(cls.getSimpleName().substring(0, cls.getSimpleName().length() - clsNameSuffix.length()));
```

Comment:
Before this change, the following didn't apply the actual config:
https://github.com/quarkusio/quarkus/blob/2b4496718ec46836e2775f669c7c8c5f970ccaa1/core/runtime/src/main/java/io/quarkus/runtime/logging/LoggingSetupRecorder.java#L73-L74
because `quarkus.log-build-time.min-level` was looked up, which obviously doesn't exist.